serde-support/src/main/java/io/micronaut/serde/support/serializers/CustomizedMapSerializer.java

```diff
@@ -100,37 +162,80 @@ public void serialize(Encoder encoder, EncoderContext context, Argument<? extend
                         }
                     }
                 }
 
+                private Serializer<? super K> getKeySerializer(EncoderContext context, K k) throws SerdeException {
+                    if (keyGeneric == null || !keyGeneric.getType().equals(k.getClass())) {
+                        keyGeneric = (Argument<K>) Argument.of(k.getClass());
+                        internalKeySerializer = findKeySerializer(context, keyGeneric);
+                    }
+                    return internalKeySerializer;
+                }
+
+                private Serializer<? super V> getValueSerializer(EncoderContext context, V v) throws SerdeException {
+                    if (valueGeneric == null || !valueGeneric.getType().equals(v.getClass())) {
+                        valueGeneric = (Argument<V>) Argument.of(v.getClass());
+                        internalValSerializer = context.findSerializer(valueGeneric).createSpecific(context, valueGeneric);
+                    }
+                    return internalValSerializer;
+                }
+
+                @Override
+                public boolean isAbsent(EncoderContext context, Map<K, V> value) {
+                    return value == null;
+                }
+
                 @Override
                 public boolean isEmpty(EncoderContext context, Map<K, V> value) {
-                    return CollectionUtils.isEmpty(value);
+                    if (CollectionUtils.isEmpty(value)) {
+                        return true;
+                    }
+                    if (includeContent != SerdeConfig.SerInclude.ALWAYS) {
+                        for (V v : value.values()) {
+                            try {
+                                Serializer<? super V> valueSerializer = getValueSerializer(context, v);
+                                switch (includeContent) {
+                                    case NON_ABSENT:
+                                        if (!valueSerializer.isAbsent(context, v)) {
+                                            return false;
+                                        }
+                                        break;
+                                    case NON_EMPTY:
+                                        if (!valueSerializer.isEmpty(context, v)) {
+                                            return false;
+                                        }
+                                        break;
+                                    case NON_NULL:
+                                        if (v != null) {
+                                            return false;
+                                        }
+                                        break;
+                                    default:
+                                        return false;
+                                }
+                            } catch (SerdeException e) {
+                                throw new RuntimeException(e);
```

Comment:
need to throw a more specific exception here

Reply:
Fixed

Reply:
What I mean was a more specific exception type. If this is a configuration error maybe `ConfigurationException`?

Reply:
I have added a sneaky throw so the original exception is thrown withou changes